sync/src/types.rs

```diff
@@ -762,7 +767,22 @@ impl SyncShared {
             HeaderView::new(header.clone(), total_difficulty)
         };
 
-        header_view.build_skip(|hash| self.get_header_view(hash));
+        let snapshot = Arc::clone(&self.shared.snapshot());
+        header_view.build_skip(
+            |hash| self.get_header_view(hash),
+            |number, current| {
+                // shortcut to return an ancestor block
+                if current.number() <= snapshot.tip_number()
```

Comment:
I think this condition is wrong, it will cause the skip table construction to almost read RocksDB.

I think the limit that must be added is `(current.number () >>= 1)> number`